buzzmobile/tests/unit/process/test_bearing.py

```diff
@@ -16,20 +16,16 @@ def test_distance(self):
         assert np.isclose(calculate_directions.get_distance(fix1, fix2), 2517000, rtol=.01)
 
 
-@with_roscore
-class TestBearingNode(unittest.TestCase):
-
+class TestBearingNode(RosTest):
     @with_launch_file('buzzmobile', 'test_params.launch')
     @launch_node('buzzmobile', 'bearing.py')
     def test_bearing_node(self):
-        with mock_pub('/fix', NavSatFix, queue_size=None) as fix_node:
+        with mock_pub('/fix', NavSatFix, queue_size=0) as fix_node:
             with check_topic('/buzzmobile/bearing', Float64) as ct:
                 # send mock data
                 fix_node.send(NavSatFix(None, None, 33.636700, -84.427863, None, None, None))
                 fix_node.send(NavSatFix(None, None, 39.029128, -111.838257, None, None, None))
 
                 # check the output from the node
-                await(ct.wait_for_message())
```

Comment:
i guess the real issue is there is no way to timeout. I can believe that tests aren't flaky anymore but if there's something weird going on and a msg was published before a listener started listening, then it'll hang forever.

I'm not saying your code is wrong in this respect. I like this solution (as gross as reading from stdin is), but I think we should still have some way to timeout and display a message ("timeout, tests can be flaky: try running once again and if it still doesnt work, investigate")